src/conmon.c

```diff
@@ -1616,23 +1617,35 @@ int main(int argc, char *argv[])
 		exit_status = get_exit_status(container_status);
 	}
 
-	if (opt_exit_dir) {
-		_cleanup_free_ char *status_str = g_strdup_printf("%d", exit_status);
-		_cleanup_free_ char *exit_file_path = g_build_filename(opt_exit_dir, opt_cid, NULL);
-		if (!g_file_set_contents(exit_file_path, status_str, -1, &err))
-			nexitf("Failed to write %s to exit file: %s", status_str, err->message);
-	}
-
 	/*
 	 * Podman injects some fd's into the conmon process so that exposed ports are kept busy while
 	 * the container runs.  Close them before we notify the container exited, so that they can be
 	 * reused immediately.
 	 */
-	for (fd = 3;; fd++) {
-		if (fd == sync_pipe_fd || fd == attach_pipe_fd || fd == dev_null_r || fd == dev_null_w)
-			continue;
-		if (close(fd) < 0 && errno == EBADF)
-			break;
+	fdsdir = opendir("/proc/self/fd");
+	if (fdsdir != NULL) {
+		int fd;
+		int dfd = dirfd(fdsdir);
+		struct dirent *next;
+
+		for (next = readdir(fdsdir); next; next = readdir(fdsdir)) {
```

Comment:
I would prefer to declare next here instead of above. feels more idiomatic

Reply:
I am fine with doing the change, but the current version follows more the rest of the `conmon.c` file where we don't use any C99 for loop (in facts we are using only `//` comments from C99)

Reply:
Hm okay i'd be fine with leaving it then, and instead update it all together in a future PR